### PR TITLE
relationals: Fix vector component accessor selection

### PR DESCRIPTION
**Pull request by svenvh**
It seems the intent was to use `.xyzw` vector component accessors in
the kernel whenever `vecSize` is 2, 3, or 4.  The condition was wrong
(i.e., always true for any value of `vecSize`), causing it to always
create a kernel with numeric (`.s0123...`) vector component accessors.

The restriction on the use of `.xyzw` for 8 and 16-component vectors
has been lifted a while ago (see
https://github.com/KhronosGroup/OpenCL-Docs/issues/549) so we do not
need this check at all.

Also, fix the `vecSize` argument of two calls of `get_order_string`
when getting the order string for the "in" kernel argument.  Pass
`inVecSize` in these cases, not `outVecSize`.

Signed-off-by: Sven van Haastregt <sven.vanhaastregt@arm.com>

### CODE REVIEW

File: test_conformance/relationals/test_shuffles.cpp
Comment (EwanC) [inline test_conformance/relationals/test_shuffles.cpp:220]: Would say that `if (vecSize < 2 && vecSize > 4)` is slightly more readable, but it's a bit of a personal preference.

**Comment by bashbaug** [comment]
Should this condition be removed entirely?

https://github.com/KhronosGroup/OpenCL-Docs/issues/549

File: test_conformance/relationals/test_shuffles.cpp
Comment (svenvh) [reply test_conformance/relationals/test_shuffles.cpp:220]: I went for the minimal fix; would have applied your suggestion but following Ben's comment I believe we can drop this altogether.

**Comment by svenvh** [comment]
> Should this condition be removed entirely?
> 
> [KhronosGroup/OpenCL-Docs#549](https://github.com/KhronosGroup/OpenCL-Docs/issues/549)

Nice catch, I think we can.  I'll update the patch to remove the `if` altogether.

**Comment by svenvh** [comment]
> I'll update the patch to remove the if altogether.

Done.  I then noticed that the `vecSize` arguments were wrong in some cases, so have fixed that in the same commit.

**Review by EwanC — DISMISSED** [review]

**Comment by lakshmih** [comment]
Qualcomm would like more time to review this.

**Review by lakshmih — APPROVED** [review]
Looks good

Sreelakshmi Haridas Maruthur
Qualcomm

File: test_conformance/relationals/test_shuffles.cpp
Comment (bashbaug) [inline test_conformance/relationals/test_shuffles.cpp:220]: I think if we really want to test all vector sizes then we should remove this `vecSize > 4` condition also.  We can do this safely because we've already checked for indices greater than three, above.

(Aside: you didn't break this either, but it would be nice to remove the incorrect second "randomly choose upper and lower case" comment below, because the .xyzw selection is exclusively lower-case.)

**Review by bashbaug — APPROVED** [review]
Mostly LGTM, this is a definite improvement.  One more possible comment to consider.

File: test_conformance/relationals/test_shuffles.cpp
Comment (svenvh) [reply test_conformance/relationals/test_shuffles.cpp]: > I think if we really want to test all vector sizes then we should remove this vecSize > 4 condition also.

Yes that seems safe; I've made the change and tested it.

> (Aside: you didn't break this either, but it would be nice to remove the incorrect second "randomly choose upper and lower case" comment below, because the .xyzw selection is exclusively lower-case.)

Yes I noticed that misleading comment too and already have a followup change to tidy up the comments in this function.  I'll push that after this PR has landed.

**Review by bashbaug — APPROVED** [review]

**Review by EwanC — APPROVED** [review]

**Comment by bashbaug** [comment]
Has two approvals and another thumbs up, merging.

**Pull request merged**
